### DISCUSSION TITLE
Validating entity fields and joins in blaze-persistence queries similar to JPA Named Queries

### DISCUSSION THREAD

**Discussion by bajce** [Blazebit/blaze-persistence#1780] [Q&A] [answered]
I have a question about using blaze-persistence and ensuring the validity of queries, specifically in comparison to the JPA Named Queries.

In JPA, Named Queries are validated at server startup. If there are any invalid ones, the server won't start, allowing us to catch and rectify any issues immediately. When using blaze-persistence, however, it appears that entity fields and joins are specified as strings. This makes me wonder: Is there any mechanism to ensure these fields and joins are valid and existing at some point, similar to how it's done for Named Queries in JPA?

For instance, when using `jakarta.persistence.criteria`, we can take advantage of generated JPAMetaModels. This way, if an entity field changes (gets removed or renamed), the code won't compile if that field is used in a criteria query. This allows us to quickly identify and adjust these queries when needed.

Is there a similar feature or method in blaze-persistence that allows us to check and validate entity fields and joins at compile-time or startup-time, ensuring that all queries remain valid and up-to-date with entity changes?

**Comment by beikov** [comment]
>Is there any mechanism to ensure these fields and joins are valid and existing at some point, similar to how it's done for Named Queries in JPA?

The Hibernate JPA Metamodel generator generates static metamodel classes that also contain string constants for every entity field, so you could use that to ensure you are referring to the correctly named field.

i.e. 

```java
cb.where(MyEntity_.ID).eq(id)
```

**Reply by bajce** [reply]
@beikov Thank you for your response.

I'm using the generated string constants from the metamodel for entity fields where applicable but this is only practical for the first level of selections. When it comes to joins, I need to assign a name to the joined entity variable.
As a workaround, I created a utility function that concatenates the given variable name with the constant from the metamodel.
So, I was wondering if there's an alternative approach in blaze-persistence that can provide a more comprehensive and direct solution for validating entity fields and joins, beyond just the first level of selections.

Thank you!

**Reply by beikov** [reply, marked as answer]
There is an open issue for generating a custom static metamodel which you can watch if you want: https://github.com/Blazebit/blaze-persistence/issues/132
